sway-lib-std/src/intrinsics.sw

```diff
@@ -10,3 +10,39 @@ pub fn is_reference_type<T>() -> bool {
 pub fn size_of<T>() -> u64 {
     __size_of::<T>()
 }
+
+/// Returns the size of a value in bytes.
+pub fn size_of_val<T>(val: T) -> u64 {
+    __size_of_val::<T>(val)
+}
+
+/// Returns the address of the given value.
+pub fn addr_of<T>(val: T) -> u64 {
+    // TODO: Replace with intrinsic: https://github.com/FuelLabs/sway/issues/855
+    if !__is_reference_type::<T>() {
+        // std::revert not available here
+        asm() {
+            rvrt zero;
+        }
+    }
+    asm(ptr: val) {
+        ptr: u64
+    }
+}
+
+/// Copies bytes from src to dst.
+pub fn copy(dst: u64, src: u64, size: u64) {
+    // TODO: Replace with intrinsic: https://github.com/FuelLabs/sway/issues/855
+    asm(dst: dst, src: src, size: size) {
+        mcp dst src size;
+    };
+}
+
+/// Determines whether the raw bytes at two points of memory are equal.
+pub fn raw_eq(first: u64, second: u64, len: u64) -> bool {
+    // TODO: Replace with intrinsic: https://github.com/FuelLabs/sway/issues/855
+    asm(first: first, second: second, len: len, result) {
+        meq result first second len;
+        result: bool
+    }
+}
```

Comment:
These aren't really intrinsics, they're just low-level memory operations.

Reply:
My reasoning was that they are in `std::intrinsics` in Rust std. If I should put these somewhere else I'm open to suggestions.

Reply:
Oh, you already edited. I can put these into `mem`, yeah. Should I keep the Rust naming? Maybe `std::mem::cpy`/`std::mem::cmp`, or `std::mem::memcpy`/`std::mem::memcmp`?

Reply:
Like pretty much all ASM blocks in the std lib I think they should become intrinsics eventually, in which case they won't need to exists as lib functions at all.  But until then `std::mem` is good.

We don't anticipate adding anything more to `intrinsics.sw`, it's there just to make `__size_of` look a bit nicer.

Reply:
Maybe they're intrinsics in Rust (probably), but in our case they're not so it would be weird to put them under `intrinsics`.

The suggested names aren't bad, but maybe `memeq` would be clearer since unlike C's memcmp it's a Boolean equal-or-not.

Reply:
I think just `eq` is an OK name -- `mem::eq()`, rather than `raw_eq`.

Reply:
Naming?

Reply:
Whoops. Just seeing the comments.